lib/ui/window.dart

```diff
@@ -730,54 +755,130 @@ class Window {
+  /// [PictureRecorder.endRecording] function on the [PictureRecorder] to obtain
+  /// the final [Picture] that represents the issued graphical operations.
   ///
-  /// The framework registers with this callback and updates the layout
-  /// appropriately.
+  /// Next, create a [SceneBuilder], and add the [Picture] to it using
+  /// [SceneBuilder.addPicture]. With the [SceneBuilder.build] method you can
+  /// then obtain a [Scene] object, which you can display to the user via this
+  /// [render] function.
   ///
   /// See also:
   ///
-  ///  * [WidgetsBindingObserver], for a mechanism at the widgets layer to
-  ///    register for notifications when this is called.
-  ///  * [MediaQuery.of], a simpler mechanism for the same.
-  VoidCallback/*?*/ get onMetricsChanged => _onMetricsChanged;
-  VoidCallback/*?*/ _onMetricsChanged;
-  Zone/*!*/ _onMetricsChangedZone = Zone.root;
+  ///  * [SchedulerBinding], the Flutter framework class which manages the
+  ///    scheduling of frames.
+  ///  * [RendererBinding], the Flutter framework class which manages layout and
+  ///    painting.
+  void render(Scene/*!*/ scene) => platformDispatcher.render(scene, this);
+}
+
+/// A top-level platform window displaying a Flutter layer tree drawn from a
+/// [Scene].
+///
+/// The current list of all Flutter views for the application is available from
+/// `WidgetsBinding.instance.platformDispatcher.views`. Only views that are of
+/// type [FlutterWindow] are top level platform windows.
+///
+/// There is also a [PlatformDispatcher.instance] singleton object in `dart:ui`
+/// if `WidgetsBinding` is unavailable, but we strongly advise avoiding a static
+/// reference to it. See the documentation for [PlatformDispatcher.instance] for
+/// more details about why it should be avoided.
+///
+/// See also:
+///
+///  * [PlatformDispatcher], which manages the current list of [FlutterView]
+///    (and thus [FlutterWindow]) instances.
+class FlutterWindow extends FlutterView {
+  FlutterWindow._({Object windowId, this.platformDispatcher})
+      : _windowId = windowId;
+
+  /// The opaque ID for this view.
+  final Object _windowId;
+
+  @override
+  final PlatformDispatcher platformDispatcher;
+
+  /// The configuration of this view.
+  @override
+  ViewConfiguration get viewConfiguration {
+    assert(platformDispatcher._viewConfigurations.containsKey(_windowId));
+    return platformDispatcher._viewConfigurations[_windowId];
+  }
+}
+
+/// A [FlutterWindow] that includes access to setting callbacks and retrieving
+/// properties that reside on the [PlatformDispatcher].
+///
+/// It is the type of the `WidgetsBinding.instance.window` singleton and the
+/// legacy global [window] singleton.
+///
+/// This class provides backward compatibility with code that was written before
+/// Flutter supported multiple top level windows. To modify or retrieve these
+/// properties, new code should use `WidgetsBinding.instance.platformDispatcher`.
```

Comment:
If we don't want people to use an API, we shouldn't provide it. If it's only for legacy use, let's @deprecate it and sunset it.

I wouldn't do that though. I would just merge PlatformDispatcher into SingletonFlutterWindow and call it Window. The single-window case is the common case for most apps.